### PR TITLE
Management Server

### PR DESCRIPTION
**Pull request by Cashmaney**
Don't have to merge everything, but this is the current build running on testnet

### CODE REVIEW

File: src/worker/controller/actions/connectivity/GetPeers.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    const callback = params.callback;
+    let length = 0;
+    let error = null;
+    try {
+      length = this._controller.engNode().getConnectedPeers().length;
+    } catch (err) {
+      error = err;
+    }
+    callback(error, length);
+    //this._controller.execCmd(constants.NODE_NOTIFICATIONS.GET_PEERS, {
```
Comment (lenak25) [inline src/worker/controller/actions/connectivity/GetPeers.js]: Please remove the comment

File: package.json [hunk 2]
```diff
@@ -72,12 +72,17 @@
   },
   "devDependencies": {
     "axios": "^0.18.1",
+    "babel-eslint": "^10.0.1",
```
Comment (lenak25) [inline package.json]: What are the new dev dependencies?

File: src/cli/cli_app.js [hunk 3]
```diff
@@ -429,6 +429,9 @@ class CLI {
         status: { url: this._statusUrl }
       };
     }
+
+    nodeConfig.extraConfig.mgmt = { mgmtBase: { } };
```
Comment (lenak25) [inline src/cli/cli_app.js]: Please add a flag to the CLI that will manage the management server (same as for the web server, to allow initializing the worker without it)

**Review by lenak25 — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (Cashmaney) [reply package.json]: Stuff that eslint needs to work with the the eslint/prettier configuration

Comment (assafmo) [reply package.json]: We don't need any of this. Please remove all eslint related dependencies 😊

**Review by assafmo — DISMISSED** [review]

**Comment by Cashmaney** [comment]
@assafmo eslint/webstorm actually detects prettier rules properly now (with the additional dependencies)

See: https://prettier.io/docs/en/integrating-with-linters.html

File: src/cli/cli_app.js [hunk 4]
```diff
@@ -320,6 +322,12 @@ class CLI {
       .option("--health [value]", "start a service for health check and status queries", port => {
         this._healthCheckPort = port;
       })
+      .option("--no-management-server", "Define port number for internal management API", port => {
```
Comment (lenak25) [inline src/cli/cli_app.js]: Copy paste :)

File: src/worker/handlers/ManagementServer.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
+const express = require("express");
+const http = require("http");
+const EventEmitter = require("events").EventEmitter;
+const constants = require("../../common/constants");
+const WEB_SERVER_CONSTANTS = constants.WEB_SERVER_CONSTANTS;
+const GET_PEERS_CONST = constants.NODE_NOTIFICATIONS.GET_PEERS;
+
+const mgmtActions = [
```
Comment (lenak25) [inline src/worker/handlers/ManagementServer.js:8]: You can add unregister also now

File: src/worker/controller/actions/sync/ReceiveAllPipelineAction.js [hunk 6]
```diff
@@ -30,10 +30,10 @@ class ReceiveAllPipelineAction {
 
     try {
       // Compare between the local state and Ethereum
-      const { missingList, excessList } = await this._controller.asyncExecCmd(
-        NODE_NOTIFY.IDENTIFY_MISSING_STATES_FROM_REMOTE,
-        { cache: cache }
-      );
+      const {
```
Comment (lenak25) [inline src/worker/controller/actions/sync/ReceiveAllPipelineAction.js:33]: Please revert this change

**Review by lenak25 — DISMISSED** [review]

**Review by lenak25 — APPROVED** [review]
